lib/solidus_jwt/devise_strategies/refresh_token.rb

```diff
@@ -0,0 +1,49 @@
+module SolidusJwt
+  module DeviseStrategies
+    class RefreshToken < Devise::Strategies::Base
+      def valid?
+        valid_grant_type? && valid_params?
+      end
+
+      def authenticate!
+        resource = SolidusJwt::Token.find_by(auth_hash)
+        return fail!(:invalid) if resource.nil? || resource.user.nil?
+
+        block = Proc.new do
+          # If we honor then mark the refresh token as stale for one time use
+          resource.honor? && resource.update_columns(active: false)
+        end
+
+        if resource.user.valid_for_authentication?(&block)
```

Comment:
This is more of a stylistic change but I think I would do

```
return fail!(:invalid) unless resource.user.valid_for_authentication?(&block)

success!(resource.user)
```

so the fall through was successful rather than a fail but again just a style thing, this works great.